### PR TITLE
Add a tuftool update example

### PR DESCRIPTION
**Pull request by nealmcb**
*Issue #, if available:*

Related to discussion on #1

*Description of changes:*

Add a tuftool update example ;)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nealmcb** [comment]
I tried to add an actual update (download) also, but had this error:
 
```
$ tuftool download --root "${ROOT}" -t file:///$WRK/tuf-repo -m file:///$WRK/tuf-repo/metadata .
Metadata error: Failed to fetch file:///srv/s/rust/tough/neal_ignore/tuftool-example/tuf-repo/metadata/timestamp.json:
  Error creating HTTP request for 'file:///srv/s/rust/tough/neal_ignore/tuftool-example/tuf-repo/metadata/timestamp.json':
  builder error for url (file:///srv/s/rust/tough/neal_ignore/tuftool-example/tuf-repo/metadata/timestamp.json):
  URL scheme is not allowed
```

Is it trying to say I can't use a `file://` url there?  If so, I don't understand why.

File: tuftool/README.md [hunk 1]
```diff
@@ -90,10 +90,26 @@ tuftool create \
...
+   --key "${WRK}/keys/root.pem" \
+   --incoming-metadata file:///$WRK/tuf-repo/metadata \
+   --targets-expires 'in 3 weeks' \
+   --targets-version 1 \
+   --snapshot-expires 'in 3 weeks' \
+   --snapshot-version 1 \
+   --timestamp-expires 'in 1 week' \
+   --timestamp-version 1 \
+   --outdir "${WRK}/tuf-repo" \
+   --metadata-url file:///$WRK/tuf-repo/metadata
```
Comment (srgothi92) [inline tuftool/README.md]: if you are modifying or making changes to targets then must provide targets directory path using `--add-targets` or `-t`.
 `--incoming-metadata` option is only for delegated targets.
You are creating a new version for targets/snapshot/timestamp during update, therefore version numbers will be `2` 
```suggestion
tuftool update  \
   --root "${ROOT}" \
   --key "${WRK}/keys/root.pem" \
   --add-targets  "${WRK}/input" \
   --targets-expires 'in 3 weeks' \
   --targets-version 2 \
   --snapshot-expires 'in 3 weeks' \
   --snapshot-version 2 \
   --timestamp-expires 'in 1 week' \
   --timestamp-version 2 \
   --outdir "${WRK}/tuf-repo" \
   --metadata-url file:///$WRK/tuf-repo/metadata
```

File: tuftool/README.md [hunk 2]
```diff
@@ -90,10 +90,26 @@ tuftool create \
 ls "${WRK}/tuf-repo/metadata"
 # and you can see our signed repository's targets here:
 ls "${WRK}/tuf-repo/targets"
+
+# Finally, the part you've all been waiting for: pushing a secure update!
```
Comment (srgothi92) [inline tuftool/README.md]:
```suggestion

### Update TUF Repo
```

File: tuftool/README.md [hunk 3]
```diff
@@ -90,10 +90,26 @@ tuftool create \
...
+
```
Comment (srgothi92) [inline tuftool/README.md:95]:
```suggestion
# update targets file
echo "1.1" > "${WRK}/input/1.txt"
```

File: tuftool/README.md [hunk 4]
```diff
@@ -90,10 +90,26 @@ tuftool create \
...
+echo "1.1" > "${WRK}/input/1.txt"
+
+tuftool update   --root "${ROOT}" \
```
Comment (srgothi92) [inline tuftool/README.md]:
```suggestion
# finally, update tuf repo!
tuftool update  --root "${ROOT}" \
```

**Review by srgothi92 — COMMENTED** [review]
Thanks, for helping us improve our documentation.

`tuftool download` only supports remote `metadata-url`.  You would have to publish your repo if you would like to try that.

File: tuftool/README.md [hunk 5]
```diff
@@ -90,10 +90,28 @@ tuftool create \
 ls "${WRK}/tuf-repo/metadata"
 # and you can see our signed repository's targets here:
 ls "${WRK}/tuf-repo/targets"
+
+# update targets file
+echo "1.1" > "${WRK}/input/1.txt"
+### Update TUF Repo
```
Comment (nealmcb) [inline tuftool/README.md:94]:
```suggestion

### Update TUF Repo
```

File: tuftool/README.md [hunk 6]
```diff
@@ -90,10 +90,29 @@ tuftool create \
...
 # and you can see our signed repository's targets here:
 ls "${WRK}/tuf-repo/targets"
+
+# update targets file
+echo "1.1" > "${WRK}/input/1.txt"
+
+### Update TUF Repo
+
+echo "1.1" > "${WRK}/input/1.txt"
+# finally, update tuf repo!
```
Comment (nealmcb) [inline tuftool/README.md]:
```suggestion

# Change one of the target files
echo "1.1" > "${WRK}/input/1.txt"

# finally, update tuf repo!
```

File: tuftool/README.md [hunk 7]
```diff
@@ -90,10 +90,31 @@ tuftool create \
 ls "${WRK}/tuf-repo/metadata"
 # and you can see our signed repository's targets here:
 ls "${WRK}/tuf-repo/targets"
+
+# update targets file
+echo "1.1" > "${WRK}/input/1.txt"
+
+### Update TUF Repo
```
Comment (nealmcb) [inline tuftool/README.md]:
```suggestion
### Update TUF Repo

# update targets file
echo "1.1" > "${WRK}/input/1.txt"
```

File: tuftool/README.md [hunk 8]
```diff
@@ -90,10 +90,31 @@ tuftool create \
 ls "${WRK}/tuf-repo/metadata"
 # and you can see our signed repository's targets here:
 ls "${WRK}/tuf-repo/targets"
+
+### Update TUF Repo
+
+# update targets file
+echo "1.1" > "${WRK}/input/1.txt"
+
```
Comment (nealmcb) [inline tuftool/README.md]:
```suggestion
```

File: tuftool/README.md (hunk 1 above)
Comment (nealmcb) [reply tuftool/README.md]: Done.
This was my first shot working with github's new inline suggestion interface. Interesting! And a bit awkward. It seemed to think the suggestions overlapped, or my intermediate changes to spacing did or something.  So this suggestion was "outdated" but I did it separately.

Now we have a bunch of tiny new commits which add some confusion. I wish I could get github to combine multiple suggestions into a single commit in a way that was easy to see them and edit them all at once.

**Comment by srgothi92** [comment]
There is a Github [blog](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/incorporating-feedback-in-your-pull-request), which tells you how to batch multiple suggestion and make a single commit. But, we prefer single commit for a change and add all PR review changes to the same commit, to avoid unnecessary merge conflicts.
Can you squash all your commits to a single commit ?

**Comment by nealmcb** [comment]
Sure - I know how to squash commits. Done.

I was just noting that the new github-specific interface review suggestion=>commit UI isn't as handy as I'd like.
Also note that you can add a "squash and merge" option to the repo's choices for PRs in this repo.
 
Thanks,

**Review by srgothi92 — APPROVED** [review]

**Review by webern — APPROVED** [review]

**Comment by iliana** [comment]
> Is it trying to say I can't use a `file://` url there? If so, I don't understand why.

...

> `tuftool download` only supports remote `metadata-url`. You would have to publish your repo if you would like to try that.

#222 :)

**Pull request merged**